src/ninja.cc

```diff
@@ -348,6 +349,96 @@ int NinjaMain::ToolGraph(const Options* options, int argc, char* argv[]) {
   return 0;
 }
 
+void ToolInputsProcessNodeDeps(Node* node, DepsLog* deps_log) {
+  // Print all of this node's deps from the deps log. This often includes files
+  // that are not known by the node's input edge.
+  if (deps_log->IsDepsEntryLiveFor(node)) {
+    DepsLog::Deps* deps = deps_log->GetDeps(node);
+    if (deps != NULL) {
+      for (int i = 0; i < deps->node_count; ++i) {
+        Node* dep = deps->nodes[i];
+        if (!dep->UsedAsDep()) {
+          dep->MarkUsedAsDep();
+          printf("%s\n", dep->path().c_str());
```

Comment:
`puts` will append a newline character for you:
http://www.cplusplus.com/reference/cstdio/puts/